src/Apps/Auction/Routes/__tests__/ConfirmBid.test.tsx

```diff
@@ -93,189 +102,220 @@ describe("Routes/ConfirmBid", () => {
     jest.resetAllMocks()
   })
 
-  it("successfully places a bid and redirect to artwork page", async () => {
-    const env = setupTestEnv()
-    const page = await env.buildPage()
+  describe("for registered users", () => {
+    it("allows the user to place a bid without agreeing to terms", async done => {
+      const env = setupTestEnv()
+      const page = await env.buildPage()
 
-    env.mutations.useResultsOnce(createBidderPositionSuccessful)
+      env.mutations.useResultsOnce(createBidderPositionSuccessful)
+      mockBidderPositionQuery.mockResolvedValue(
+        confirmBidBidderPositionQueryWithPending
+      )
 
-    await page.agreeToTerms()
-    await page.submitForm()
+      await page.submitForm()
 
-    expect(env.mutations.mockFetch).toHaveBeenCalledWith(
-      expect.objectContaining({
-        name: "ConfirmBidCreateBidderPositionMutation",
-      }),
-      {
-        input: {
-          artwork_id: "artworkslug",
-          max_bid_amount_cents: 5000000,
-          sale_id: "saleslug",
-        },
-      }
-    )
+      expect(env.mutations.mockFetch).toHaveBeenCalledWith(
+        expect.objectContaining({
+          name: "ConfirmBidCreateBidderPositionMutation",
+        }),
+        {
+          input: {
+            artwork_id: "artworkslug",
+            max_bid_amount_cents: 5000000,
+            sale_id: "saleslug",
+          },
+        }
+      )
 
-    expect(window.location.assign).toHaveBeenCalledWith(
-      `https://example.com/auction/${
-        ConfirmBidQueryResponseFixture.artwork.saleArtwork.sale.id
-      }/artwork/${ConfirmBidQueryResponseFixture.artwork.id}`
-    )
-  })
+      expect(mockBidderPositionQuery).toHaveBeenCalledTimes(1)
+      expect(mockBidderPositionQuery.mock.calls[0][1]).toEqual({
+        bidderPositionID: "positionid",
+      })
 
-  it("tracks a success event to Segment including Criteo info", async () => {
-    const env = setupTestEnv()
-    const page = await env.buildPage()
-    env.mutations.useResultsOnce(createBidderPositionSuccessful)
-
-    await page.agreeToTerms()
-    await page.submitForm()
-
-    expect(mockPostEvent).toBeCalledWith({
-      action_type: AnalyticsSchema.ActionType.ConfirmBidSubmitted,
-      context_page: AnalyticsSchema.PageName.AuctionConfirmBidPage,
-      auction_slug: "saleslug",
-      artwork_slug: "artworkslug",
-      bidder_id: "bidderid",
-      bidder_position_id: "positionid",
-      sale_id: "saleid",
-      user_id: "my-user-id",
-      order_id: "bidderid",
-      products: [
-        {
-          product_id: "artworkid",
-          quantity: 1,
-          price: 50000,
-        },
-      ],
+      mockBidderPositionQuery.mockReset()
+      mockBidderPositionQuery.mockResolvedValue(
+        confirmBidBidderPositionQueryWithWinning
+      )
+
+      setTimeout(() => {
```

Comment:
This is a bit regrettable but our test setup is a bit heavier than I initially anticipated, so I couldn't figure out how to use the Jest timer in tandem with the page object. But there must be a better way than `setTimeout`.

cc @ds300

Reply:
I think it should be possible to use jest's mock timers with `createTestEnv`. Do you remember what issues you ran into? Happy to pair on this if you have some spare minutes soon!